### ISSUE TITLE
[8.12] TestStandaloneUpgradeWithGPGFallback tests fails unconditionally because the wrong 8.12.1 artifact is downloaded

### ISSUE THREAD

**Issue by cmacknz** [elastic/elastic-agent#4317]
Reproducing the content from https://github.com/elastic/elastic-agent/pull/4243#issuecomment-1944752650

W are downloading the 8.12.1 artifact from staging

```
fetcher_artifact.go:311: Completed downloading artifact from https://staging.elastic.co/8.12.1-4dcfae35/downloads/beats/elastic-agent/elastic-agent-8.12.1-linux-arm64.tar.gz
    fetcher_artifact.go:265: Downloading artifact from https://staging.elastic.co/8.12.1-4dcfae35/downloads/beats/elastic-agent/elastic-agent-8.12.1-linux-arm64.tar.gz.sha512
    fetcher_artifact.go:311: Completed downloading artifact from https://staging.elastic.co/8.12.1-4dcfae35/downloads/beats/elastic-agent/elastic-agent-8.12.1-linux-arm64.tar.gz.sha512
    fetcher_artifact.go:265: Downloading artifact from https://staging.elastic.co/8.12.1-4dcfae35/downloads/beats/elastic-agent/elastic-agent-8.12.1-linux-arm64.tar.gz.asc
    fetcher_artifact.go:311: Completed downloading artifact from https://staging.elastic.co/8.12.1-4dcfae35/downloads/beats/elastic-agent/elastic-agent-8.12.1-linux-arm64.tar.gz.asc
    fixture.go:261: Extracting artifact elastic-agent-8.12.1-linux-arm64.tar.gz to /tmp/TestStandaloneUpgradeWithGPGFallback2510475544/002
```

However the agent is downloading the official 8.12.1 release from artifacts.elastic.co because it doesn't have staging as a source URI.

For whatever reason the hash is different between staging and artifacts, I am not sure why this is because the build summary of the artifact we actually released matches what is on artifacts.elastic.co.

```
ubuntu@upright-mullet:~$ ./elastic-agent-8.12.1-linux-arm64/elastic-agent version --binary-only --yaml
binary:
  version: 8.12.1
  commit: 9db5527750c7b3510cbedca152590c24444eba2b
  build_time: 2024-02-01T18:42:33Z
  snapshot: false
ubuntu@upright-mullet:~$ ./staging/elastic-agent-8.12.1-linux-arm64/elastic-agent version --binary-only --yaml
binary:
  version: 8.12.1
  commit: 0015eb6095376c180afc99a12bddb2a983460142
  build_time: 2024-02-07T06:34:36Z
  snapshot: false
```

When we download artifacts for use in tests we need to prefer artifacts.elastic.co over staging.elastic.co.

**Comment by cmacknz** [comment]
We are going to skip the affected tests while we add a solution for this problem.

**Comment by cmacknz** [comment]
PR disabling the affected tests:

- [ ] https://github.com/elastic/elastic-agent/pull/4318

**Comment by cmacknz** [comment]
Assigning this to @pchila since he has a draft PR attempting to address this: https://github.com/elastic/elastic-agent/pull/4330

**Comment by pchila** [comment]
Fix merged on 8.12, backported on 8.13 and main branches
